x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/execution/search/extractor/MetricAggExtractor.java

```diff
@@ -41,22 +48,31 @@
     private final String name;
     private final String property;
     private final String innerKey;
-    private final boolean isDateTimeBased;
+    private final DataType dataType;
     private final ZoneId zoneId;
 
-    public MetricAggExtractor(String name, String property, String innerKey, ZoneId zoneId, boolean isDateTimeBased) {
+    public MetricAggExtractor(String name, String property, String innerKey, ZoneId zoneId, @Nullable DataType dataType) {
         this.name = name;
         this.property = property;
         this.innerKey = innerKey;
-        this.isDateTimeBased = isDateTimeBased;
+        this.dataType = dataType;
         this.zoneId = zoneId;
     }
 
     MetricAggExtractor(StreamInput in) throws IOException {
         name = in.readString();
         property = in.readString();
         innerKey = in.readOptionalString();
-        isDateTimeBased = in.readBoolean();
+        if (in.getVersion().onOrAfter(Version.V_7_13_0)) {
+            String typeName = in.readOptionalString();
+            if (typeName != null) {
+                dataType = SqlDataTypes.fromTypeName(typeName);
+            } else {
+                dataType = null;
+            }
+        } else {
+            dataType = in.readBoolean() ? DATETIME : null;
+        }
```

Comment:
As a reminder this code is actually NOT needed in master since that's 8.0 which means all other nodes during an upgrade will be at least 7.13.
That is, this code should be removed in master and be kept only in 7.x

Reply:
Good point, removed.